tests/test_end_to_end.py

```diff
@@ -55,11 +58,52 @@ def test_model_info(client):
     ]
 
 
-def test_http(client):
-    response = client.get("/species/{}".format('ECOLX'))
+def test_species(client):
+    response = client.get("/species/ECOLX")
     assert response.status_code == 200
+    assert len(response.json) > 0
+
+
+def test_simulate_wrong_id(client):
+    response = client.post("/models/wrong_id/simulate", json={'message': {}})
+    assert response.status_code == 404
+
+
+def test_simulate_no_message(client):
+    response = client.post("/models/iJO1366/simulate", json={})
+    assert response.status_code == 400
+
+
+def test_simulate_infeasible(client):
+    response = client.post("/models/iJO1366/simulate", json={'message': MESSAGE_FLUXES_INFEASIBLE})
+    result = response.json
+    assert result['fluxes']['ATPM'] == pytest.approx(100)
+
+
+def test_simulate_modify(monkeypatch, client):
```

Comment:
If I understand the logic of the test correctly then this could also be reformulated into a parametrized test. It would mean a bigger re-organization, though.